### PR TITLE
#3525 - Remove Overawards Deductions from PT e-Cert

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Removed the `applyOverawardsDeductionsStep` from part-time e-Cert process.
- Created the below E2E tests.
  - `queue-consumers`: Should create an e-Cert with no overaward deductions when the student has overawards.
  - `workers`: Should generate an overaward and deduct grants already paid for a part-time application when a reassessment happens and the student is entitled to less money.

### CODE REVIEW

File: sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-calculation/part-time-calculation-process.ts [hunk 1]
```diff
@@ -58,7 +56,6 @@ export class PartTimeCalculationProcess extends ECertCalculationProcess {
   protected calculationSteps(): ECertProcessStep[] {
     return [
       this.validateDisbursementPartTimeStep,
-      this.applyOverawardsDeductionsStep,
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-calculation/part-time-calculation-process.ts:61]: 👍

File: sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/save-disbursement-schedules-payloads.ts [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
+ * - `createSecondDisbursement`: indicates if the second schedule should be created. Default false.
+ * - `firstDisbursementAwards`: optional values to be used. If not provided, default values will be used.
+ * - `secondDisbursementAwards`: optional values to be used. If not provided, default values will be used.
+ * @returns disbursement schedule payload.
+ */
+export function createFakeSaveDisbursementSchedulesPayload(options: {
+  assessmentId: number;
+  createSecondDisbursement?: boolean;
+  firstDisbursementAwards?: DisbursementValue[];
+  secondDisbursementAwards?: DisbursementValue[];
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/save-disbursement-schedules-payloads.ts:28]: Minor. we can use options here rather.

File: sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/save-disbursement-schedules-payloads.ts [hunk 3]
```diff
@@ -1,24 +1,41 @@
...
+ * Create the disbursement schedule payload expected to be received from the workflow.
+ * @param options creation options.
+ * - `assessmentId`: assessment to save the schedules.
+ * - `createSecondDisbursement`: indicates if the second schedule should be created. Default false.
+ * - `firstDisbursementAwards`: optional values to be used. If not provided, default values will be used.
+ * - `secondDisbursementAwards`: optional values to be used. If not provided, default values will be used.
+ * @returns disbursement schedule payload.
+ */
+export function createFakeSaveDisbursementSchedulesPayload(options: {
+  assessmentId: number;
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/controllers/disbursement/_tests_/e2e/save-disbursement-schedules-payloads.ts:25]: Minor. the assessment could be outside options if that is required.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts [hunk 4]
```diff
@@ -1102,6 +1102,104 @@ describe(
...
+          disbursementValues: {
+            valueCode: "CSLP",
+          },
+        },
+      });
+      expect(schedule.disbursementScheduleStatus).toBe(
+        DisbursementScheduleStatus.Sent,
+      );
+      const [cslpDisbursementValue] = schedule.disbursementValues;
+      expect(cslpDisbursementValue.effectiveAmount).toEqual(1000);
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-part-time-process-integration.scheduler.e2e-spec.ts:1191]: 👍

**Review by dheepak-aot — APPROVED** [review]
Good Job! One minor comment which is not a blocker.

Thanks for adding the worker e2e test.

**Review by lewischen-aot — APPROVED** [review]
LGTM. Nice work 👍

**Pull request merged**